src/components/App.tsx

```diff
@@ -7757,6 +7774,30 @@ class App extends React.Component<AppProps, AppState> {
     this.setState({ suggestedBindings });
   }
 
+  public setSelection(elements: readonly NonDeletedExcalidrawElement[]) {
+    const selectedElementIds: { [id: string]: true } = {};
+    const selectedGroupIds: { [id: string]: true } = {};
+
+    elements.forEach((ele) => {
+      if (ele.groupIds.length) {
+        selectedElementIds[ele.id] = true;
+        ele.groupIds.forEach((id) => {
+          selectedGroupIds[id] = true;
+        });
+      }
+      // exclude bound text elements as we don't mark them as selected when
+      // container is selected unless in group
+      else if (!isBoundToContainer(ele)) {
+        selectedElementIds[ele.id] = true;
+      }
+    });
+
+    this.setState({
+      previousSelectedElementIds: this.state.selectedElementIds,
+      selectedElementIds,
```

Comment:
replace with `selectedElementIds: makeNextSelectedElementIds(selectedElementIds, prevState)` to keep the object stable when appliacble

Reply:
actually also for `selectedGroupIds` (use the same function, we'll need to rename later)

Reply:
Actually I just realized we no more need this API at all since we are using `addElementsFromPasteOrLibrary` hence I am removing this API altogether